awspricing/offers.py

```diff
@@ -212,7 +214,8 @@ def ondemand_hourly(self,
             region=region,
             capacity_status=capacity_status
         )
-        term = self._offer_data['terms']['OnDemand'][sku]
+        offer = [x for x in self._offer_data if x['product']['sku'] == sku][0]
```

Comment:
Could we use the reverse SKU mapping here instead?

Reply:
Do you mean search the offers by `self._reverse_sku` instead of the `sku`?

Reply:
Yep - iterating through `self._offer_data` each time would be quite inefficient. Is the data you're looking for already in `self._reverse_sku`?

Reply:
So getting the sku in the previous line uses self._reverse_sku; then with that sku we want to search through the offer data to actually get the pricing terms. `self._reverse_sku` looks like `{'c4.large|Linux|Shared|No License required|NA|US East (N. Virginia)|Used': '4C7N4APU9GEUZ6H6'}` from a test, so don't think the price information is available there

Reply:
Ah, OK. It's been a while so I have forgotten a lot of this :) Is there a way we can make this a dict lookup instead of iterating through the entire `offer_data` each time? I think this would be a significant performance regression

Reply:
Gotcha, this makes sense. Will refactor this to be a dict lookup